scripts/run_e2e_tests_test.py

```diff
@@ -360,6 +361,108 @@ def test_start_tests_in_debug_mode(self) -> None:
         with self.swap_mock_set_constants_to_default:
             run_e2e_tests.main(args=['--debug_mode'])
 
+    def test_skip_firebase_and_datastore_emulator_when_not_in_emulator_mode(
+        self
+    ) -> None:
+        self.exit_stack.enter_context(self.swap_with_checks(
+            common, 'is_oppia_server_already_running', lambda *_: False))
+        self.exit_stack.enter_context(self.swap_with_checks(
+            run_e2e_tests, 'install_third_party_libraries', lambda _: None,
+            expected_args=[(False,)]))
+        self.exit_stack.enter_context(self.swap_with_checks(
+            build, 'build_js_files', lambda *_, **__: None,
+            expected_args=[(True,)]))
+        self.exit_stack.enter_context(self.swap_with_checks(
+            servers, 'managed_elasticsearch_dev_server', mock_managed_process))
+        self.exit_stack.enter_context(self.swap_with_checks(
+            servers, 'managed_dev_appserver', mock_managed_process))
+        self.exit_stack.enter_context(self.swap_with_checks(
+            servers, 'managed_redis_server', mock_managed_process))
+        self.exit_stack.enter_context(self.swap_with_checks(
+            servers, 'managed_portserver', mock_managed_process))
+
+        self.exit_stack.enter_context(self.swap_with_checks(
+            servers, 'managed_webdriverio_server', mock_managed_process,
+            expected_kwargs=[
+                {
+                    'suite_name': 'full',
+                    'chrome_version': None,
+                    'dev_mode': True,
+                    'mobile': False,
+                    'sharding_instances': 3,
+                    'debug_mode': False,
+                    'stdout': subprocess.PIPE,
+                },
+            ]))
+        self.exit_stack.enter_context(self.swap_with_checks(
+            sys, 'exit', lambda _: None, expected_args=[(0,)]))
+
+        emulator_mode_swap = self.swap(constants, 'EMULATOR_MODE', False)
+
+        with self.swap_mock_set_constants_to_default:
+            with emulator_mode_swap:
+                run_e2e_tests.main(args=[])
+
+    def test_loop_invocation_continues_while_poll_is_none(
```

Comment:
"poll is none" doesn't really make sense, maybe say "while process is still alive" and leave an inline comment in poll_mock() to explain what its simulating.

Reply:
Done